documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,158 @@
+
+    * **_ipport_**
+
+      For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.
+
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## Client Credentials Flow
+
+The [OAuth 2.0 Client Credentials](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4) grant type is used to obtain a 2-legged Access Token that does not represent a user. This grant type can only be used when no personal user data is processed, and it is only a valid option to access the CAMARA APIs for these specific scenarios.
+
+
+### Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects.
+
+How to apply the concept of purpose with scope can be found in [Section Purposes of CAMARA APIs access and user consent management](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md).
+
+## ID Token sub claim
+
+The subject identifier must be globally unique identifier and a case-sensitive ASCII string with a maximum length of 255. The sub value MUST not contain the MSISDN. Within Camara this is populated by a Pseudonymous Customer Reference (PCR) which is a system-facing unique identifier that links a User’s MSISDN/IP-Address to a ASP. It is generated by the OP.
+
+## Client Authentication
+
+This CAMARA document allows **one** client authentication method, private_key_jwt, as defined in OIDC
+[OIDC Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication)
+[JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/rfc/rfc7523.html#section-2.2)
+
+## Sender-constraint Access Token
+
+This CAMARA document RECOMMENDS sender-constraint access tokens.
+
+### DPoP Requirements for Camara Clients
+
+Camara clients SHOULD utilize DPoP as described in [RFC9449].
+
+If Camara clients use DPoP, they shall support the server provided nonce mechanism (as defined in Section 8 of [RFC9449])
+
+### DPoP Requirements for Camara Authorizatin Servers
+
+Camara Authorization Servers MUST support DPoP as described in [RFC9449].
+
+### DPoP Requirements for Resource Servers
+
+Camara Resource Servers MUST support and verify sender-constrained access tokens using DPoP as described in [RFC9449].
```

Comment:
Thanks. As I say in https://github.com/camaraproject/IdentityAndConsentManagement/issues/122#issuecomment-1945931102 this would be my recommendation to move forward:

> 1. Remove extra features that have not even been discussed yet (DPOP, PKCE) from https://github.com/camaraproject/IdentityAndConsentManagement/pull/121, to compare apples to apples. Open separate issues to discuss each new feature as separate track.
> 2. In parallel, give 1-2 weeks to gather all the issues we currently have in production due to different interpretations of the RFCs. Review whether they are solved in
https://github.com/camaraproject/IdentityAndConsentManagement/pull/113 and
https://github.com/camaraproject/IdentityAndConsentManagement/pull/121. Fix them otherwise.
> 3. Choose one option. I don't have a strong opinion as long as they are complete and reflect our current agreements + clarifications needed to avoid repeating the same issues that have caused we (CSPs) are not 100% interoperable.

> I think no harm is done by recommending better security.

Well, everything has pros and cons. DPoP is not a "small recommendation". It comes with a significant impact on Authorization Servers, Applications, and API Gateways... This is why it shouldn't come out of the blue, mixing the debate about the profile with whether DPoP is a good addition or not.

Reply:
I removed the section on sender-constraint access tokens